### PR TITLE
Upgrade dotnet sdk to 5.0.102

### PR DESCRIPTION
**Pull request by rbans96**
## Description
Upgrade dotnet sdk to 5.0.100

## Related issues
Addresses [issue #].
Couple of latest PRs are failing due to [this ](https://lgtm.com/projects/g/microsoft/fhir-server/logs/rev/pr-a35fd5980a4edfcc1aca672930bee509355c4815/lang:csharp/stage:Build%20main_268834e6fbdc6e7d167672a44adb6ef16e122902)(screenshot below) error
image

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)
+semver: feature

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
👏

File: samples/templates/default-azuredeploy.json [hunk 1]
```diff
@@ -301,7 +301,7 @@
                     "apiVersion": "2018-11-01",
                     "location": "[resourceGroup().location]",
                     "properties": {
-                        "version": "3.1.11"
+                        "version": "5.0.0"
```
Comment (brendankowitz) [inline samples/templates/default-azuredeploy.json:304]: Probably will need to change the extension above from `"AspNetCoreRuntime.3.1.x86"` too

File: src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj [hunk 2]
```diff
@@ -62,12 +62,12 @@
       <Generated Include="Features\Schema\Model\ImmutableTypes.Generated.cs">
         <Generator>ImmutableSqlModelGenerator</Generator>
         <Namespace>Microsoft.Health.Fhir.SqlServer.Features.Schema.Model</Namespace>
-        <Args>@(ImmutableSqlGeneratorInputs->'&quot;%(FullPath)&quot;', ' ')</Args>
+        <Args>@(ImmutableSqlGeneratorInputs->'"%(FullPath)"', ' ')</Args>
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj:65]: Revert the quotes

File: src/Microsoft.Health.Fhir.Stu3.Web/Microsoft.Health.Fhir.Stu3.Web.csproj [hunk 3]
```diff
@@ -10,7 +10,7 @@
     <PackageReference Include="IdentityServer4" Version="4.1.1" />
     <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.16.0" />
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.0" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.AzureKeyVault" Version="$(SdkPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.AzureKeyVault" Version="3.1.11" />
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Stu3.Web/Microsoft.Health.Fhir.Stu3.Web.csproj:13]: On nuget it says we should consider moving to https://www.nuget.org/packages/Azure.Extensions.AspNetCore.Configuration.Secrets/

File: src/Microsoft.Health.Fhir.Azure/Microsoft.Health.Fhir.Azure.csproj [hunk 4]
```diff
@@ -11,6 +11,7 @@
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.0" />
     <PackageReference Include="Microsoft.Azure.Storage.Blob" Version="11.2.2" />
     <PackageReference Include="Polly" Version="7.2.1" />
+      <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="5.0.0" />
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Azure/Microsoft.Health.Fhir.Azure.csproj:14]: $(SdkPackageVersion)

File: src/Microsoft.Health.Fhir.Stu3.Web/Microsoft.Health.Fhir.Stu3.Web.csproj (hunk 3 above)
Comment (rbans96) [reply src/Microsoft.Health.Fhir.Stu3.Web/Microsoft.Health.Fhir.Stu3.Web.csproj]: Right, Thanks!

File: src/Microsoft.Health.Fhir.Shared.Web/Program.cs [hunk 5]
```diff
@@ -26,9 +24,8 @@ public static void Main(string[] args)
                     var keyVaultEndpoint = builtConfig["KeyVault:Endpoint"];
                     if (!string.IsNullOrEmpty(keyVaultEndpoint))
                     {
-                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
-                        var keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
-                        builder.AddAzureKeyVault(keyVaultEndpoint, keyVaultClient, new DefaultKeyVaultSecretManager());
+                        var credential = new DefaultAzureCredential();
+                        builder.AddAzureKeyVault(new System.Uri(keyVaultEndpoint), credential);
```
Comment (rbans96) [inline src/Microsoft.Health.Fhir.Shared.Web/Program.cs:28]: Used [this ](https://stackoverflow.com/questions/62503022/net-core-key-vault-configuration-using-azure-security-keyvault-secrets
)as reference to move to new package Azure.Extensions.AspNetCore.Configuration.Secrets

File: build/docker/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/dotnet/core/sdk:3.1.405-bionic AS build
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build
```
Comment (rbans96) [inline build/docker/Dockerfile:1]: @brendankowitz I have updated the .net sdk image name from the [docker hub documentation](https://hub.docker.com/_/microsoft-dotnet-sdk/). I am not sure if we should find the bionic version of 5.0.100 as I couldn't find any documentation around this. Thoughts?

File: Directory.Build.props [hunk 7]
```diff
@@ -12,7 +12,7 @@
     <EnableSourceControlManagerQueries>$(EnableSourceLink)</EnableSourceControlManagerQueries>
     <!--This will target the latest patch release of the runtime released with the current SDK.  -->
     <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
-    <SdkPackageVersion>3.1.11</SdkPackageVersion>
+    <SdkPackageVersion>5.0.2</SdkPackageVersion>
```
Comment (Ivanidzo4ka) [inline Directory.Build.props:15]: >SdkPackageVersion [](start = 5, length = 17)

I would split this variable into two variables:
`<RuntimePackageVersion>5.0.0</RuntimePackageVersion>`
and 
`<AspPackageVersion>5.0.2</AspPackageVersion>`

File: Directory.Build.props [hunk 8]
```diff
@@ -12,7 +12,8 @@
     <EnableSourceControlManagerQueries>$(EnableSourceLink)</EnableSourceControlManagerQueries>
     <!--This will target the latest patch release of the runtime released with the current SDK.  -->
     <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
-    <SdkPackageVersion>3.1.11</SdkPackageVersion>
+    <AspPackageVersion>5.0.2</AspPackageVersion>
```
Comment (brendankowitz) [inline Directory.Build.props:15]: nit: AspNetPackageVersion

File: Directory.Build.props [hunk 9]
```diff
@@ -12,7 +12,8 @@
...
+    <RunTimePackageVersion>5.0.0</RunTimePackageVersion>
```
Comment (brendankowitz) [inline Directory.Build.props:16]: Nit: I don't think RunTime is capitalized: `RuntimePackageVersion`

File: Directory.Build.props (hunk 7 above)
Comment (brendankowitz) [reply Directory.Build.props]: This should be ok, each variable will be split into a separate PR by dependabot.

Comment (rbans96) [reply Directory.Build.props:15]: Sounds good! Reverted the change to split the versions as it will be taken care by dependabot.

**Comment by brendankowitz** [comment]
I think these tests will be fixed by #1650

File: Directory.Build.props (hunk 7 above)
Comment (brendankowitz) [reply Directory.Build.props:15]: Lol sorry, keep what you had (RuntimePackageVersion & AspNetPackageVersion)! Without the variables we will get multiple PRs from dependabot for packages that should probably version together

Comment (rbans96) [reply Directory.Build.props:15]: Sorry to  misunderstand, will revert the reverted :)

**Review by brendankowitz — APPROVED** [review]
I think `+semver: feature` 🚀 
Thoughts?

**Comment by rbans96** [comment]
> I think `+semver: feature` 🚀
> Thoughts?

Good point!

**Pull request merged**